satpy/readers/yaml_reader.py

```diff
@@ -386,11 +398,13 @@ def new_filehandler_instances(self, filetype_info, filename_items):
             try:
                 req_fh = self.find_required_filehandlers(requirements,
                                                          filename_info)
-            except RuntimeError:
-                logger.warning("Can't find requirements for %s", filename)
-                continue
-            except KeyError:
-                logger.warning("Missing requirements for %s", filename)
+            except (KeyError, RuntimeError) as req:
+                # Monkeypatch warnings module to omit the source line - just
+                # for this warning.
+                warnings.formatwarning = formatwarning_no_source
+                warnings.warn("Missing requirement {} for {}".format(
+                    req, filename))
+                warnings.formatwarning = formatwarning_orig
```

Comment:
Why remove the source line from the warning?

Reply:
Only the last line of the statement is included which is not very informative in this case: ``req, filename))``

Reply:
Having said that, I could of course store the message in a variable so that the warning statement fits in one line ^^ But even then it does not provide much additional information. On the other hand, the current approach is less readable/maintainable.

Reply:
"readability counts" :)

Reply:
Our maximum line width is now 120 characters so it should be fine to put this all on one line and remove the special `no_source` function.